Sming/SmingCore/Platform/Station.cpp

```diff
@@ -49,12 +49,12 @@ bool StationClass::isEnabled()
 	return wifi_get_opmode() & STATION_MODE;
 }
 
-bool StationClass::config(String ssid, String password, bool autoConnectOnStartup /* = true*/, bool save /* = true */)
+bool StationClass::config(const char * ssid, const char * password, bool autoConnectOnStartup /* = true*/, bool save /* = true */)
 {
 	station_config config = {0};
 
-	if (ssid.length() >= sizeof(config.ssid)) return false;
-	if (password.length() >= sizeof(config.password)) return false;
+	if (sizeof(ssid) >= sizeof(config.ssid)) return false;
+	if (sizeof(password) >= sizeof(config.password)) return false;
```

Comment:
should be strlen(password). sizeof (char*) is 4 bytes it does not reflect the length of the string